src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -307,6 +261,121 @@ def tool_valid(tool: str) -> bool:
             return False
         return True
 
+    @staticmethod
+    def resolve_dependencies(main_ctx: Any, sbom_path: str) -> dict[str, DependencyInfo]:
```

Comment:
I wonder why `main_ctx` is type-annotated as `Any` instead of `AnalyzeContext` here.

Reply:
@nathanwn This question is not relevant to this PR because @benmss has not authored this function and just moved it to [dependency_resolver.py](https://github.com/oracle/macaron/pull/388/files/724b3993ef12517afc94f457e4998a1667160b90#diff-c95724433501b4a68d8ffe2e35305bff56f2c26c45571f9250641511a8d48269) to keep dependency resolution functions in the same place.

Reply:
Got it.

Reply:
Hmm I don't see why the question could be irrelevant even though the change belongs to this PR. 
I guess the concerned part is whether we want to address it in this PR or not. I guess because this function wasn't authored by @benmss , it's okay for us to address in another PR (if applicable). However, I don't think there is any issue if we want to know more about the code in the first place. 
To me, this `Any` type annotation is a bit confusing and I would raise the same question if I were reviewing this PR.

Reply:
@tromai a review tip: when we review a PR we need to understand the context and why a change has happened. If a question needs to be asked outside the context of a PR, we can explicitly mention that, ask the relevant contributor, and follow up separately. For example in this case, @benmss was not the author of the code in the first place, so asking him about this code is not useful. But this question can be asked from the actual author (by mentioning that person) and follow up if needed separately.